### ISSUE TITLE
[v2] error Couldn't find match for "prebuild" in "[...]" for "https://github.com/lovell/sharp.git"

### ISSUE THREAD

**Issue by kripod** [gatsbyjs/gatsby#5440]
## Description

Since installing gatsby-plugin-sharp@2.0.0-alpha.15, I get the error message in the title during a fresh `yarn` execution without a lockfile.

### Steps to reproduce

1. Clone [gatsby-starter-modern](https://github.com/kripod/gatsby-starter-modern)
2. Run `yarn add gatsby-plugin-sharp@2.0.0-alpha.15`
3. Delete `node_modules` and `yarn.lock`
4. Execute `yarn`

### Expected result

No errors should be thrown.

### Actual result

```
yarn install v1.6.0
info No lockfile found.
[1/4] Resolving packages...
warning gatsby-plugin-sharp > imagemin-webp > cwebp-bin > bin-build > download > gulp-decompress > gulp-util@3.0.8: gulp-util is deprecated - replace it, following the guidelines at https://medium.com/gulpjs/gulp-util-ca3b1f9f9ac5
error Couldn't find match for "prebuild" in "refs/heads/master,refs/pull/101/head,refs/pull/101/merge,refs/pull/102/head,refs/pull/1023/head,refs/pull/1023/merge,refs/pull/1024/head,refs/pull/1032/head,refs/pull/1032/merge,refs/pull/1038/head,refs/pull/106/head,refs/pull/107/head,refs/pull/1077/head,refs/pull/1079/head,refs/pull/11/head,refs/pull/111/head,refs/pull/1111/head,refs/pull/1111/merge,refs/pull/1121/head,refs/pull/1142/head,refs/pull/1161/head,refs/pull/1165/head,refs/pull/1204/head,refs/pull/1208/head,refs/pull/1229/head,refs/pull/124/head,refs/pull/124/merge,refs/pull/13/head,refs/pull/13/merge,refs/pull/14/head,refs/pull/146/head,refs/pull/146/merge,refs/pull/15/head,refs/pull/156/head,refs/pull/160/head,refs/pull/161/head,refs/pull/165/head,refs/pull/166/head,refs/pull/171/head,refs/pull/173/head,refs/pull/175/head,refs/pull/18/head,refs/pull/192/head,refs/pull/194/head,refs/pull/198/head,refs/pull/20/head,refs/pull/20/merge,refs/pull/205/head,refs/pull/205/merge,refs/pull/207/head,refs/pull/211/head,refs/pull/211/merge,refs/pull/213/head,refs/pull/22/head,refs/pull/244/head,refs/pull/25/head,refs/pull/255/head,refs/pull/263/head,refs/pull/266/head,refs/pull/271/head,refs/pull/271/merge,refs/pull/272/head,refs/pull/272/merge,refs/pull/280/head,refs/pull/282/head,refs/pull/286/head,refs/pull/287/head,refs/pull/288/head,refs/pull/291/head,refs/pull/302/head,refs/pull/302/merge,refs/pull/303/head,refs/pull/305/head,refs/pull/306/head,refs/pull/307/head,refs/pull/308/head,refs/pull/308/merge,refs/pull/309/head,refs/pull/316/head,refs/pull/334/head,refs/pull/334/merge,refs/pull/335/head,refs/pull/335/merge,refs/pull/351/head,refs/pull/359/head,refs/pull/361/head,refs/pull/368/head,refs/pull/374/head,refs/pull/374/merge,refs/pull/38/head,refs/pull/384/head,refs/pull/384/merge,refs/pull/385/head,refs/pull/385/merge,refs/pull/39/head,refs/pull/393/head,refs/pull/393/merge,refs/pull/397/head,refs/pull/399/head,refs/pull/402/head,refs/pull/404/head,refs/pull/408/head,refs/pull/419/head,refs/pull/425/head,refs/pull/425/merge,refs/pull/426/head,refs/pull/426/merge,refs/pull/439/head,refs/pull/440/head,refs/pull/440/merge,refs/pull/443/head,refs/pull/448/head,refs/pull/456/head,refs/pull/465/head,refs/pull/473/head,refs/pull/479/head,refs/pull/480/head,refs/pull/484/head,refs/pull/484/merge,refs/pull/49/head,refs/pull/492/head,refs/pull/496/head,refs/pull/497/head,refs/pull/498/head,refs/pull/499/head,refs/pull/500/head,refs/pull/501/head,refs/pull/503/head,refs/pull/503/merge,refs/pull/504/head,refs/pull/51/head,refs/pull/511/head,refs/pull/513/head,refs/pull/516/head,refs/pull/516/merge,refs/pull/519/head,refs/pull/52/head,refs/pull/52/merge,refs/pull/520/head,refs/pull/520/merge,refs/pull/540/head,refs/pull/55/head,refs/pull/550/head,refs/pull/550/merge,refs/pull/569/head,refs/pull/569/merge,refs/pull/570/head,refs/pull/581/head,refs/pull/593/head,refs/pull/593/merge,refs/pull/596/head,refs/pull/596/merge,refs/pull/61/head,refs/pull/61/merge,refs/pull/617/head,refs/pull/617/merge,refs/pull/622/head,refs/pull/623/head,refs/pull/644/head,refs/pull/644/merge,refs/pull/654/head,refs/pull/654/merge,refs/pull/656/head,refs/pull/663/head,refs/pull/675/head,refs/pull/676/head,refs/pull/68/head,refs/pull/685/head,refs/pull/691/head,refs/pull/732/head,refs/pull/735/head,refs/pull/738/head,refs/pull/742/head,refs/pull/742/merge,refs/pull/748/head,refs/pull/754/head,refs/pull/754/merge,refs/pull/760/head,refs/pull/769/head,refs/pull/769/merge,refs/pull/783/head,refs/pull/790/head,refs/pull/790/merge,refs/pull/791/head,refs/pull/798/head,refs/pull/798/merge,refs/pull/813/head,refs/pull/814/head,refs/pull/824/head,refs/pull/825/head,refs/pull/825/merge,refs/pull/828/head,refs/pull/83/head,refs/pull/857/head,refs/pull/87/head,refs/pull/87/merge,refs/pull/888/head,refs/pull/906/head,refs/pull/906/merge,refs/pull/91/head,refs/pull/91/merge,refs/pull/913/head,refs/pull/915/head,refs/pull/919/head,refs/pull/924/head,refs/pull/924/merge,refs/pull/925/head,refs/pull/925/merge,refs/pull/94/head,refs/pull/940/head,refs/pull/940/merge,refs/pull/945/head,refs/pull/954/head,refs/pull/954/merge,refs/pull/956/head,refs/pull/956/merge,refs/pull/960/head,refs/pull/974/head,refs/pull/974/merge,refs/pull/975/head,refs/pull/976/head,refs/pull/977/head,refs/pull/986/head,refs/pull/986/merge,refs/pull/990/head,refs/pull/990/merge,refs/pull/991/head,refs/pull/991/merge,refs/pull/992/head,refs/tags/v0.0.1,refs/tags/v0.0.2,refs/tags/v0.0.3,refs/tags/v0.0.4,refs/tags/v0.0.5,refs/tags/v0.0.6,refs/tags/v0.0.8,refs/tags/v0.0.9,refs/tags/v0.1.0,refs/tags/v0.1.1,refs/tags/v0.1.2,refs/tags/v0.1.3,refs/tags/v0.1.4,refs/tags/v0.1.5,refs/tags/v0.1.6,refs/tags/v0.1.7,refs/tags/v0.1.8,refs/tags/v0.10.0,refs/tags/v0.10.1,refs/tags/v0.11.0,refs/tags/v0.11.1,refs/tags/v0.11.2,refs/tags/v0.11.3,refs/tags/v0.11.4,refs/tags/v0.12.0,refs/tags/v0.12.1,refs/tags/v0.12.2,refs/tags/v0.13.0,refs/tags/v0.13.1,refs/tags/v0.14.0,refs/tags/v0.14.1,refs/tags/v0.15.0,refs/tags/v0.15.1,refs/tags/v0.16.0,refs/tags/v0.16.1,refs/tags/v0.16.2,refs/tags/v0.17.0,refs/tags/v0.17.1,refs/tags/v0.17.2,refs/tags/v0.17.3,refs/tags/v0.18.0,refs/tags/v0.18.1,refs/tags/v0.18.2,refs/tags/v0.18.3,refs/tags/v0.18.4,refs/tags/v0.19.0,refs/tags/v0.19.1,refs/tags/v0.2.0,refs/tags/v0.20.0,refs/tags/v0.20.1,refs/tags/v0.20.2,refs/tags/v0.3.0,refs/tags/v0.4.0,refs/tags/v0.4.1,refs/tags/v0.4.2,refs/tags/v0.5.0,refs/tags/v0.5.1,refs/tags/v0.5.2,refs/tags/v0.6.0,refs/tags/v0.6.1,refs/tags/v0.6.2,refs/tags/v0.7.0,refs/tags/v0.7.1,refs/tags/v0.7.2,refs/tags/v0.8.0,refs/tags/v0.8.1,refs/tags/v0.8.2,refs/tags/v0.8.3,refs/tags/v0.8.4,refs/tags/v0.9.0,refs/tags/v0.9.1,refs/tags/v0.9.2,refs/tags/v0.9.3" for "https://github.com/lovell/sharp.git".
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

### Environment
* Gatsby version (`npm list gatsby`): 2.0.0-alpha.32
* gatsby-cli version (`gatsby --version`): 2.0.0-alpha.32
* Node.js version: 8.11.1
* Operating System: Windows 10 x64

### File contents (if changed)

See https://github.com/kripod/gatsby-starter-modern/tree/d2ce864348782014214842da3e3b484a14b5a83c.

**Comment by KyleAMathews** [comment]
I get this consistently too. For some reason it will work with npm. The reason I suspect is there's some possible dependency that relies on a former canary release that pointed to the temp git branch for sharp that now doesn't exist.

**Comment by nihgwu** [comment]
that's because `gatsby-remark-images` depends on an range version of `gatsby-plugin-sharp` https://github.com/gatsbyjs/gatsby/blob/v2/packages/gatsby-remark-images/package.json#L12, which will resolved to canary version `2.0.0-alpha.e9c3027b
` which is quite out of date, so I alwasy use `resolutions` to specify the exact version of `gatsby-plugin-sharp`

**Comment by pieh** [comment]
For now you will probably need to lock to `2.0.0-alpha.15` (without `^` carret) or use `next`. As @nihgwu noticed and I had similiar findings earlier ( #5170 ) with different package, yarn seems to think that `alpha.[hash]` is higher version than `alpha.15`.

**Comment by m-allanson** [comment]
Fixed in #5443

**Comment by qdhaiqiang** [comment]
git ls-remote --tags --heads https://github.com/lovell/sharp.git
